sdk/spring/spring-cloud-azure-feature-management/src/main/java/com/azure/spring/cloud/feature/management/filters/TimeWindowFilter.java

```diff
@@ -35,38 +43,37 @@ public TimeWindowFilter() {
      */
     @Override
     public boolean evaluate(FeatureFilterEvaluationContext context) {
-        String start = (String) context.getParameters().get(TIME_WINDOW_FILTER_SETTING_START);
-        String end = (String) context.getParameters().get(TIME_WINDOW_FILTER_SETTING_END);
+        final Map<String, Object> parameters = context.getParameters();
+        final Object recurrenceObject = parameters.get(TIME_WINDOW_FILTER_SETTING_RECURRENCE);
+        if (recurrenceObject != null) {
+            final Map<String, Object> recurrenceParameters = (Map<String, Object>) recurrenceObject;
+            final Object patternObj = recurrenceParameters.get(RecurrenceConstants.RECURRENCE_PATTERN);
+            if (patternObj != null) {
+                FeatureFilterUtils.updateValueFromMapToList((Map<String, Object>) patternObj, RecurrenceConstants.RECURRENCE_PATTERN_DAYS_OF_WEEK);
+            }
+        }
 
-        ZonedDateTime now = ZonedDateTime.now();
+        final TimeWindowFilterSettings settings = OBJECT_MAPPER.convertValue(context.getParameters(), TimeWindowFilterSettings.class);
+        final ZonedDateTime now = ZonedDateTime.now();
 
-        if (!StringUtils.hasText(start) && !StringUtils.hasText(end)) {
+        if (settings.getStart() == null && settings.getEnd() == null) {
             LOGGER.warn("The {} feature filter is not valid for feature {}. It must specify either {}, {}, or both.",
```

Comment:
@mrm9084 Shall we throw exception here? 
In the validations of recurrence, I just throw exception, because if there're invalid value for the enum type, it's easier to throw exception. Maybe we need to keep consistent?